`isinstance-second-argument-not-valid-type` false positive for union types with `None`

### Bug description

#8206 does not fix the issue #8205 if the union type contains `None`.
```
# pylint: disable=missing-docstring,invalid-name
a = 1
isinstance(a, None | int)  # raises W1116
```

### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
************* Module test
test.py:3:0: W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```


### Expected behavior

no warnings

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8424.false_positive
+++ b/doc/whatsnew/fragments/8424.false_positive
@@ -1,0 +1,3 @@
+Fix false positive for isinstance-second-argument-not-valid-type when union types contains None.
+
+Closes #8424

--- a/pylint/checkers/typecheck.py
+++ b/pylint/checkers/typecheck.py
@@ -37,6 +37,7 @@ from pylint.checkers.utils import (
     is_mapping,
     is_module_ignored,
     is_node_in_type_annotation_context,
+    is_none,
     is_overload_stub,
     is_postponed_evaluation_enabled,
     is_super,
@@ -798,8 +799,9 @@ def _is_c_extension(module_node: InferenceResult) -> bool:
 def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
     # Return True if we are sure that arg is not a type
     if PY310_PLUS and isinstance(arg, nodes.BinOp) and arg.op == "|":
-        return _is_invalid_isinstance_type(arg.left) or _is_invalid_isinstance_type(
-            arg.right
+        return any(
+            _is_invalid_isinstance_type(elt) and not is_none(elt)
+            for elt in (arg.left, arg.right)
         )
     inferred = utils.safe_infer(arg)
     if not inferred:
@@ -812,9 +814,10 @@ def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
     if isinstance(inferred, astroid.Instance) and inferred.qname() == BUILTIN_TUPLE:
         return False
     if PY310_PLUS and isinstance(inferred, bases.UnionType):
-        return _is_invalid_isinstance_type(
-            inferred.left
-        ) or _is_invalid_isinstance_type(inferred.right)
+        return any(
+            _is_invalid_isinstance_type(elt) and not is_none(elt)
+            for elt in (inferred.left, inferred.right)
+        )
     return True
 
 

--- a/tests/functional/i/isinstance_second_argument_py310.py
+++ b/tests/functional/i/isinstance_second_argument_py310.py
@@ -1,13 +1,17 @@
-'''Tests for invalid isinstance with compound types'''
+"""Tests for invalid isinstance with compound types"""
 
 # True negatives
 isinstance(0, int | str)
 isinstance(0, int | int | int)
 isinstance(0, int | str | list | float)
 isinstance(0, (int | str) | (list | float))
+isinstance(0, int | None)
+isinstance(0, None | int)
 
 IntOrStr = int | str
 isinstance(0, IntOrStr)
+IntOrNone = int | None
+isinstance(0, IntOrNone)
 ListOrDict = list | dict
 isinstance(0, (float | ListOrDict) | IntOrStr)
 

--- a/tests/functional/i/isinstance_second_argument_py310.txt
+++ b/tests/functional/i/isinstance_second_argument_py310.txt
@@ -1,3 +1,3 @@
-isinstance-second-argument-not-valid-type:15:0:15:22::Second argument of isinstance is not a type:INFERENCE
-isinstance-second-argument-not-valid-type:16:0:16:28::Second argument of isinstance is not a type:INFERENCE
-isinstance-second-argument-not-valid-type:18:0:18:24::Second argument of isinstance is not a type:INFERENCE
+isinstance-second-argument-not-valid-type:19:0:19:22::Second argument of isinstance is not a type:INFERENCE
+isinstance-second-argument-not-valid-type:20:0:20:28::Second argument of isinstance is not a type:INFERENCE
+isinstance-second-argument-not-valid-type:22:0:22:24::Second argument of isinstance is not a type:INFERENCE

